docs/TerraformEnvironmentStage.md

```diff
@@ -0,0 +1,44 @@
+## [TerraformEnvironmentStagePlugin](../src/TerraformEnvironmentStagePlugin.groovy)
```

Comment:
Should this be `TerraformEnvironmentStage` and not `TerraformEnvironmentStagePlugin`?

Reply:
so I struggled with that. I had just Stage before but to follow the pattern other plugins I used the plugin of it since the plugin calls EnvironmentStage

Reply:
Yeah, that's part of the reason why it's undocumented.  It's "odd" behavior.

But that convention aside, the documentation is explaining the `.withEnv`, and `.withGlobalEnv`, and neither of those methods exist on the code that's being linked to by the documentation.

Reply:
The README has a section [Additional Pipeline Stages](https://github.com/manheim/terraform-pipeline#additional-pipeline-stages).  What if instead of calling that "Additional", what if it were simply "Pipeline Stages".  At which point, we can add `TerraformEnvironmentStage` to the list, and link to your doc.  

Would that make for a better convention?

Reply:
The above would also solve the problem of making this documentation "find-able".

Reply:
that works. I'll update the PR

Reply:
updated